internal/schema/schema.go

```diff
@@ -770,3 +832,30 @@ func parseFieldsDef(l *common.Lexer) FieldList {
 	}
 	return fields
 }
+
+// interfaceDependancies searches through the tree of interface relations
+// to build a slice of all the names that should be implemented on the
+// given interface
+//
+// TODO: Would be more efficient not to search every interface multiple
+// times.
+//
+// NOTE: https://spec.graphql.org/draft/#sel-FAHbhBHCAACGB35P
+func interfaceDependancies(iface Interface) []string {
+	var search func(requiredNames []string, iface Interface) []string
+
+	search = func(requiredNames []string, iface Interface) []string {
+		for _, f := range iface.Interfaces {
+			if f == nil {
+				continue
+			}
```

Comment:
Can this happen? Is it possible that one of the `iface.Interfaces` is `nil`?

Reply:
No, good point, ifaces should never be nil in the current implementation. Just used to checking pointer types, is there a reason all the types are using pointers or can I change `Schema.Interfaces` to a non-pointer type and remove this check?

Reply:
I'm not sure what the reason is since I haven't written this code but my guess is that the author didn't want to copy structs by value for performance reasons. On the other hand, interfaces are not supposed to be huge. But we never know how people use it. So let's leave it consistent with all other types.

I'd prefer if both the `interfaceDependancies` and `search` functions accept a reference. Then at the beginning of the `search` function check if the passed pointer to the `Interface` struct is `nil` and if yes, then panic. This is never supposed to happen.